### DISCUSSION TITLE
raise requests.exceptions.ConnectionError(Failed to establish a new connection: [WinError 10048])  when I request interface 15860 times

### DISCUSSION THREAD

**Discussion by bfgf52** [fastapi/fastapi#6264] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
async def root():
    return {"message": "Hello World"}
```

### Description

I start the server with command uvicorn main:app.
My client code is 
```
import requests

cnt = 0
while True:
    requests.get('http://127.0.0.1:8000')
    print(cnt)
    cnt += 1
```
when cnt == 15858,it raise error
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000016264E85FC8>: Failed to establish a new connection: [WinError 10048]

### Operating System

Windows

### FastAPI Version

fastapi==0.75.2

### Python Version

3.7.13

**Comment by iudeen** [comment]
Where are you running your app and your client? Same PC?

**Comment by JarroVGIT** [comment, marked as answer]
This is not a FastAPI issue (that is handling it all fine), but a limitation on how `requests` works. If you put heavy load on it, you better use `requests.Session` like so:

```python
import requests

cnt = 0
s = requests.Session()
while True:
    s.get('http://127.0.0.1:8000')
    print(cnt)
    cnt += 1
```
This is running fine, haven't hit the limit yet :)

EDIT: I cut it off at 1.5mln requests, it proved its point 😜

**Comment by bfgf52** [comment]
Thanks for your reply, it works well!
